### ISSUE TITLE
doesn't warn about unused mutability of self but does with other method inputs

### ISSUE THREAD

**Issue by noahbadoa** [rust-lang/rust-analyzer#12929]
For example the rust analyzer warns only complains about the argument x and not the argument self here. Despite neither being mutated.

```
struct test{i : i32}

impl test{
    pub fn method2(&mut self, mut x : i32){
        println!("{}", x)
    }
}
```

**Comment by Veykril** [comment]
Mutability warnings are rustc diagnostics, not rust-analyzer ones.

**Comment by flodiebold** [comment]
Also, this is not a bug: These are different kinds of `mut`. The `&mut self` is a mutable (unique) reference, whereas the `mut x` just makes the binding mutable. There are good reasons to want a `&mut` reference even if you don't mutate it, so the compiler doesn't warn about that. There are no good reasons to mark a binding `mut` when you don't change it.
